components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -299,12 +299,16 @@ abstract class AbstractFetchDownloadService : Service() {
      * Be aware that you need to pass [latestUIStatus] as [DownloadJobState.status] can be modified
      * from another thread, causing inconsistencies in the ui.
      */
-    private fun updateDownloadNotification(latestUIStatus: DownloadJobStatus, download: DownloadJobState) {
+    @VisibleForTesting
+    internal fun updateDownloadNotification(latestUIStatus: DownloadJobStatus, download: DownloadJobState) {
         val notification = when (latestUIStatus) {
             ACTIVE -> DownloadNotification.createOngoingDownloadNotification(context, download)
             PAUSED -> DownloadNotification.createPausedDownloadNotification(context, download)
-            COMPLETED -> DownloadNotification.createDownloadCompletedNotification(context, download)
             FAILED -> DownloadNotification.createDownloadFailedNotification(context, download)
+            COMPLETED -> {
+                addToDownloadSystemDatabaseCompact(download.state)
```

Comment:
So this will be always called regardless of whether or not the OS version is less than Android Q? It looks addToDownloadSystem has only an if{} but not an else, and is only actionable for versions less than Android Q; is it possible to check this outside the function to reduce unnecessary function calls? (I'm not sure what the best practices are here but just wanted to point this out)

Reply:
Good catch, normally on Android when something does actions that are backward support compatibility the functions or clases has the  **compat** word on it to indicate that support older versions for example [ContextCompat](https://developer.android.com/reference/androidx/core/content/ContextCompat) and [ActivityCompat](https://developer.android.com/reference/androidx/core/app/ActivityCompat). I put it on the function as I didn't want to enlarge the `updateDownloadNotification` function, also I added the note on the description of the function :)

Reply:
Oh then ultra-nit but should it be renamed to Compat instead of Compact? I think that had me a little confused, I didn't catch the reference/

Reply:
No problem, I already updated it :)

Reply:
Awesome, you changed it! Thank you 👍